data/net/minecraft/world/level/levelgen/synth/PerlinNoise.mapping

```diff
@@ -1,24 +1,37 @@
 CLASS net/minecraft/world/level/levelgen/synth/PerlinNoise
+	COMMENT This class generates multiple octaves of perlin noise. Each individual octave is an instance of {@link net.minecraft.world.level.levelgen.synth.ImprovedNoise}
+	COMMENT Mojang uses the term 'Perlin' to describe octaves or fBm (Fractal Brownian Motion) noise.
 	METHOD <init> (Lnet/minecraft/world/level/levelgen/RandomSource;Lcom/mojang/datafixers/util/Pair;)V
 		ARG 1 randomSource
+		ARG 2 amplitudes
 	METHOD <init> (Lnet/minecraft/world/level/levelgen/RandomSource;Lcom/mojang/datafixers/util/Pair;Ljava/util/function/LongFunction;)V
 		ARG 1 randomSource
+		ARG 2 amplitudes
+		ARG 3 randomSourceFactory
 	METHOD <init> (Lnet/minecraft/world/level/levelgen/RandomSource;Lit/unimi/dsi/fastutil/ints/IntSortedSet;)V
 		ARG 1 randomSource
+		ARG 2 octaves
 	METHOD <init> (Lnet/minecraft/world/level/levelgen/RandomSource;Lit/unimi/dsi/fastutil/ints/IntSortedSet;Ljava/util/function/LongFunction;)V
 		ARG 1 randomSource
+		ARG 2 octaves
+		ARG 3 randomSourceFactory
 	METHOD <init> (Lnet/minecraft/world/level/levelgen/RandomSource;Ljava/util/List;)V
 		ARG 1 randomSource
+		ARG 2 octaves
 	METHOD <init> (Lnet/minecraft/world/level/levelgen/RandomSource;Ljava/util/stream/IntStream;)V
 		ARG 1 randomSource
+		ARG 2 octaves
 	METHOD create (Lnet/minecraft/world/level/levelgen/RandomSource;ILit/unimi/dsi/fastutil/doubles/DoubleList;)Lnet/minecraft/world/level/levelgen/synth/PerlinNoise;
 		ARG 0 randomSource
+		ARG 1 octaves
 		ARG 2 amplitudes
 	METHOD create (Lnet/minecraft/world/level/levelgen/RandomSource;I[D)Lnet/minecraft/world/level/levelgen/synth/PerlinNoise;
 		ARG 0 randomSource
+		ARG 1 octaves
 		ARG 2 amplitudes
 	METHOD getOctaveNoise (I)Lnet/minecraft/world/level/levelgen/synth/ImprovedNoise;
-		ARG 1 octaveIndex
+		COMMENT Gets a single octave of Perlin noise.
+		ARG 1 octave
```

Comment:
```suggestion
		ARG 1 index
```

Reply:
I disagree. I'd named all instances of similar parameters (`IntStream`) octaves, or amplitudes (`DoubleList`) matching the mojang field name. This is accessing an individual octave, *by* octave, the specific array index really is meaningless here.

Especially as the array is being indexed in reverse, i.e. `this.noiseLevels[this.noiseLevels.length - 1 - octave]`

Reply:
In that case I should mention that there was another method somewhere that used `index` as opposed to octave for this same method, I believe.